src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Metadata/Models/ContentPartFieldDefinitionExtensions.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement.Metadata.Settings;
+
+namespace OrchardCore.ContentManagement.Metadata.Models
+{
+    public static class ContentPartFieldDefinitionExtensions
+    {
+        public static bool IsNamedPart(this ContentPartFieldDefinition fieldDefinition)
+            => fieldDefinition.PartDefinition.IsReusable() && fieldDefinition.ContentTypePartDefinition.Name != fieldDefinition.PartDefinition.Name;
+
+        /// <summary>
+        /// Returns full field name.
+        /// </summary>
+        /// <param name="fieldDefinition">The <see cref="ContentPartFieldDefinition" />.</param>
+        /// <returns></returns>
+        public static string GetFullName(this ContentPartFieldDefinition fieldDefinition)
+            => $"{fieldDefinition.PartDefinition.Name}-{fieldDefinition.Name}";
+
+        /// <summary>
+        /// Returns unique full field name.
+        /// </summary>
+        /// <param name="fieldDefinition">The <see cref="ContentPartFieldDefinition" />.</param>
+        /// <returns></returns>
+        public static string GetUniqueFullName(this ContentPartFieldDefinition fieldDefinition)
+            =>  $"{fieldDefinition.ContentTypePartDefinition.Name}-{fieldDefinition.Name}";
+
+
+        public static IEnumerable<string> GetAdditionalClasses(this ContentPartFieldDefinition fieldDefinition, string prefix, Func<string, string> transform)
```

Comment:
Maybe rename this method to GetFieldWrapperCssClasses or something like that. You also don't need to pass transform callback here. The name names should always be transformed with the HtmlClassify helper. Also, not sure if prefix is helpful parameter. Maybe instead pass param string[] which allows you to add any additional classes. Prefix should be a fixed value in the method.

Reply:
What if you move the `GetFieldWrapperCssClasses` and `GetPartWrapperCssClasses` to the `CssOrchardHelper` class using something like this?

```
    public static IEnumerable<string> GetFieldWrapperCssClasses(this IOrchardHelper helper,  ContentTypePartDefinition partDefinition, params string[] additionalClasses)
    {

    }
```

Reply:
done